src/components/home/ProductList/index.tsx

```diff
@@ -1,17 +1,42 @@
 import type { ReactElement } from 'react'
+import { useEffect, useState } from 'react'
+import { useInView } from 'react-intersection-observer'
 import { Styled } from './styled'
 import type { ProductListProps } from './types'
 import { ProductItem } from '../ProductItem'
 
-const ProductList = ({ productList }: ProductListProps): ReactElement => {
+const ProductList = ({
+  postData,
+  hasNextPage,
+  fetchNextPage
+}: ProductListProps): ReactElement => {
+  const [isFirstRender, setIsFirstRender] = useState<boolean>(false)
+
+  const { ref: isLastPrdRef, inView } = useInView({
+    threshold: 1
+  })
+
+  useEffect(() => {
+    if (inView && hasNextPage) {
+      if (!isFirstRender) {
+        setIsFirstRender(true)
+        return
+      }
+      fetchNextPage && fetchNextPage()
+    }
+  }, [inView, hasNextPage, fetchNextPage])
```

Comment:
```suggestion
      if (!(inView && hasNextPage)) {
        return
      }

      if (!isFirstRender) {
        setIsFirstRender(true)
        return
      }
      
      fetchNextPage && fetchNextPage()
    }
  }, [inView, hasNextPage, fetchNextPage])
```
`!(inView && hasNextPage)` 이 조건을 변수로 빼고 위에서 early return처리하는 게 더 이해하기 쉬울 것 같습니다!

Reply:
넵 패치 가능 조건이었는데, 말씀주신데로 변수화 + early 리턴이 좋아보입니다! 👍  shouldFetchNextPage 로 반영 
[완료](https://github.com/price-offer/offer-fe/pull/168/commits/02a1412339ae0a5f5e5ad6532c83d618a429873e#diff-3e51e0f8d6e4568cd4669d93f9556fc6fc8e772054d2c329af0832783d363617)